dnsmasq_interface.c

```diff
@@ -1078,7 +1069,7 @@ static int add_blocked_domain_cache(struct all_addr *addr4, struct all_addr *add
 // Add a single domain to resolver's cache. This respects the configured blocking mode
 static void block_single_domain(char *domain)
 {
-	struct all_addr addr4, addr6;
+	struct all_addr addr4 = { 0 }, addr6 = { 0 };
```

Comment:
For the sake of documentation: If there are fewer initializers in a brace-enclosed list than there are elements or members of an aggregate, or fewer characters in a string literal used to initialize an array of known size than there are elements in the array (this is the case here), the remainder of the aggregate shall be initialized implicitly the same as objects that have static storage duration according to the C99 Standard.